src/shim/crypto.rs

```diff
@@ -209,6 +246,47 @@ pub fn verify_bls_sig(
     fvm_shared_latest::crypto::signature::ops::verify_bls_sig(signature, data, &addr.into())
 }
 
+/// Returns `String` error if a delegated signature is invalid.
+pub fn verify_delegated_sig(
+    signature: &[u8],
+    data: &[u8],
+    addr: &crate::shim::address::Address,
+) -> anyhow::Result<()> {
+    use super::fvm_shared_latest::{
+        address::Protocol::Delegated,
+        crypto::signature::{ops::recover_secp_public_key, SECP_SIG_LEN},
+    };
+    use crate::rpc::eth::types::EthAddress;
+    use crate::utils::encoding::keccak_256;
+
+    anyhow::ensure!(
+        addr.protocol() == Delegated,
+        "cannot validate a delegated signature against a {} address expected",
+        addr.protocol()
+    );
+
+    anyhow::ensure!(
+        signature.len() == SECP_SIG_LEN,
+        "invalid delegated signature length. Was {}, must be {}",
+        signature.len(),
+        SECP_SIG_LEN
+    );
+
+    let hash = keccak_256(data);
+    let mut sig = [0u8; SECP_SIG_LEN];
+    sig[..].copy_from_slice(signature);
```

Comment:
nit: remove second `ensure` and `mut` and replace them with some `try_into()`:

```
let sig: [u8; SECP_SIG_LEN] = signature.try_into().with_context(|| {
    format!(
        "invalid delegated signature length. Was {}, must be {}",
        signature.len(),
        SECP_SIG_LEN,
    )
})?;
```

Reply:
Done.